### PR TITLE
Policy studio - Add create / edit / delete  message flow

### PR DESCRIPTION
**Pull request by ThibaudAV**
**Issue**
https://gravitee.atlassian.net/browse/APIM-1550

**Description**

Add create / edit / delete  message flow

Next PR : same dialog but for proxy flow. Maybe i rename this dialog or not if a change only the content, we will see

**Additional context**

image

image

---
#### 🧪&nbsp;&nbsp;Gravitee.io Automatic Prerelease @gravitee/ui-policy-studio-angular
```
npm install @gravitee/ui-policy-studio-angular@7.3.0-apim-1550-crud-message-flow-c4fb521
```
```
yarn add @gravitee/ui-policy-studio-angular@7.3.0-apim-1550-crud-message-flow-c4fb521
```

---
#### 🧪&nbsp;&nbsp;Gravitee.io Automatic Prerelease @gravitee/ui-particles-angular
```
npm install @gravitee/ui-particles-angular@7.3.0-apim-1550-crud-message-flow-c4fb521
```
```
yarn add @gravitee/ui-particles-angular@7.3.0-apim-1550-crud-message-flow-c4fb521
```

---

### CODE REVIEW

File: ui-particles-angular/projects/ui-policy-studio-angular/src/lib/components/flow-form-dialog/gio-ps-flow-form-dialog.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+  it('should return false on cancel', async () => {
+    component.flow = {
+      _id: 'test-id',
+      ...fakeChannelFlow(),
+    };
+    await componentTestingOpenDialog();
+
+    const flowFormDialogHarness = await loader.getHarness(GioPolicyStudioFlowFormDialogHarness);
+    await flowFormDialogHarness.cancel();
```
Comment (loriepisicchio) [inline ui-particles-angular/projects/ui-policy-studio-angular/src/lib/components/flow-form-dialog/gio-ps-flow-form-dialog.component.spec.ts:98]: Maybe we should perform some changes here and check they are not saved on cancel ?

File: ui-particles-angular/projects/ui-policy-studio-angular/src/lib/components/flow-form-dialog/gio-ps-flow-form-dialog.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+        },
+        {
+          type: 'CONDITION',
+          condition: 'Test condition',
+        },
+      ],
+    });
+  });
+
+  it('should edit flow with any changes', async () => {
```
Comment (loriepisicchio) [inline ui-particles-angular/projects/ui-policy-studio-angular/src/lib/components/flow-form-dialog/gio-ps-flow-form-dialog.component.spec.ts:169]:
```suggestion
  it('should edit flow without any changes', async () => {
```

**Review by loriepisicchio — APPROVED** [review]

File: ui-particles-angular/projects/ui-policy-studio-angular/src/lib/components/flow-form-dialog/gio-ps-flow-form-dialog.component.spec.ts (hunk 2 above)
Comment (ThibaudAV) [reply ui-particles-angular/projects/ui-policy-studio-angular/src/lib/components/flow-form-dialog/gio-ps-flow-form-dialog.component.spec.ts:169]: haha ty

**Pull request merged**